### PR TITLE
HRTIM

### PR DESCRIPTION
**Pull request by usbalbin**
**NOTE**: This is targeted at the `staged-pac` branch which depends on the newer staged pac

### CODE REVIEW

**Comment by usbalbin** [comment]
This is by no means complete. There are a lot of missing things like listening on events from neighbor timers, burst mode, chopper etc. However I still believe it does cover quite a lot of use cases. Some of which are showed in the examples.

I am no type system ninja so there are probably *a lot* of things that can(and should?) be improved upon :)

File: src/hrtim/external_event.rs [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+            eev_input8: EevInput { _x: PhantomData },
+            eev_input9: EevInput { _x: PhantomData },
+            eev_input10: EevInput { _x: PhantomData },
+        }
+    }
+}
+
+pub struct EevInput<const N: u8> {
+    _x: PhantomData<()>,
+}
```
Comment (burrbull) [inline src/hrtim/external_event.rs:49]: With `#[non_exhaustive]`  you can make constructor private with using `PhantomData`.
```suggestion
impl EevInputs {
    pub(crate) unsafe fn new() -> Self {
        EevInputs {
            eev_input1: EevInput,
            eev_input2: EevInput,
            eev_input3: EevInput,
            eev_input4: EevInput,
            eev_input5: EevInput,
            eev_input6: EevInput,
            eev_input7: EevInput,
            eev_input8: EevInput,
            eev_input9: EevInput,
            eev_input10: EevInput,
        }
    }
}

#[non_exhaustive]
pub struct EevInput<const N: u8>;
```

Comment (usbalbin) [reply src/hrtim/external_event.rs]: Wow, I had no idea. Thanks :)

Comment (burrbull) [reply src/hrtim/external_event.rs]: This was just example. There are several other empty `PhantomData<()>`.

Comment (usbalbin) [reply src/hrtim/external_event.rs]: The others should be fixed now aswell

**Comment by usbalbin** [comment]
Closing this in favor of usbalbin/stm32-hrtim#1

**Pull request closed without merging**